src/elements/content-explorer/stories/tests/ContentExplorer-visual.stories.js

```diff
@@ -1,20 +1,195 @@
+                await userEvent.click(cancelButton);
+                expect(screen.queryByText('Are you sure you want to delete Book Sample.pdf?')).not.toBeInTheDocument();
+            },
+            {
+                timeout: SLEEP_TIMEOUT,
+            },
+        );
+    },
+};
+
+export const openRenameDialog = {
+    play: async ({ canvasElement }) => {
+        const canvas = within(canvasElement);
+
+        await waitFor(
+            async () => {
+                const moreOptionsButton = canvas.getByTestId('bce-btn-more-options');
+                await userEvent.click(moreOptionsButton);
+
+                const renameButton = screen.getByText('Rename');
+                expect(renameButton).toBeInTheDocument();
+                await userEvent.click(renameButton);
+
+                expect(screen.getByText('Please enter a new name for Book Sample:')).toBeInTheDocument();
+            },
+            {
+                timeout: SLEEP_TIMEOUT,
+            },
+        );
+    },
+};
+
+export const closeRenameDialog = {
+    play: async ({ canvasElement }) => {
+        const canvas = within(canvasElement);
+
+        await waitFor(
+            async () => {
+                const moreOptionsButton = canvas.getByTestId('bce-btn-more-options');
+                await userEvent.click(moreOptionsButton);
+
+                const renameButton = screen.getByText('Rename');
+                expect(renameButton).toBeInTheDocument();
+                await userEvent.click(renameButton);
+                expect(screen.getByText('Please enter a new name for Book Sample:')).toBeInTheDocument();
+
+                const cancelButton = screen.getByText('Cancel');
+                await userEvent.click(cancelButton);
+                expect(screen.queryByText('Please enter a new name for Book Sample:')).not.toBeInTheDocument();
+            },
+            {
+                timeout: SLEEP_TIMEOUT,
+            },
+        );
+    },
+};
+
+export const openShareDialog = {
+    play: async ({ canvasElement }) => {
+        const canvas = within(canvasElement);
+
+        await waitFor(
+            async () => {
+                const moreOptionsButton = canvas.getByTestId('bce-btn-more-options');
+                await userEvent.click(moreOptionsButton);
+
+                const dropdown = await waitFor(() => screen.getByRole('menu'));
+                const shareButton = within(dropdown).getByText('Share');
+                expect(shareButton).toBeInTheDocument();
+                await userEvent.click(shareButton);
+
+                await waitFor(() => {
+                    expect(screen.getByText('Shared Link:')).toBeInTheDocument();
+                });
+
+                const inputElement = screen.getByDisplayValue('https://example.com/share-link');
+                expect(inputElement).toBeInTheDocument();
+            },
+            {
+                timeout: SLEEP_TIMEOUT,
+            },
+        );
+    },
+};
```

Comment:
```
export const openShareDialog = {
    play: async ({ canvasElement }) => {
        const canvas = within(canvasElement);

        const moreOptionsButton = await canvas.findByTestId('bce-btn-more-options');
        await userEvent.click(moreOptionsButton);

        const dropdown = await screen.findByRole('menu');
        const shareButton = within(dropdown).getByText('Share');
        expect(shareButton).toBeInTheDocument();
        await userEvent.click(shareButton);

        expect(await screen.findByText('Shared Link:')).toBeInTheDocument();

        const inputElement = screen.getByDisplayValue('https://example.com/share-link');
        expect(inputElement).toBeInTheDocument();
    },
};
```

I tried something like this to get away from wrapping everything with a waitFor and it seems to be a cleaner solution, thought?

the `findBy` return promises which gives us the state changes in the rendering we want when going through the actions